quickstart/202-machine-learning-moderately-secure-existing-VNet/readme.md

```diff
@@ -0,0 +1,63 @@
+
+This deployment configuration specifies an [Azure Machine Learning workspace](https://docs.microsoft.com/en-us/azure/machine-learning/concept-workspace), 
+and its associated resources including Azure Key Vault, Azure Storage, Azure Application Insights and Azure Container Registry.
+
+In addition to these core services, this configuration specifies any networking components that are required to set up Azure Machine Learning
+for private network connectivity using [Azure Private Link](https://docs.microsoft.com/en-us/azure/private-link/). 
+
+This configuration describes the minimal set of resources you require to get started with Azure Machine Learning in a network-isolated set-up. This configurations assumes that you have existing network components to reuse. The [201 example](../201-machine-learning-moderately-secure/readme.md), alternatively creates new network components. 
+
+Please note that this template does not create Azure Private DNS zones. The assumption is that you have already configured Azure private DNS zones that are linked to your virtual network resources.
+
+## Resources
+
+| Terraform Resource Type | Description |
+| - | - |
+| `azurerm_resource_group` | The resource group all resources get deployed into |
+| `azurerm_application_insights` | An Azure Application Insights instance associated to the Azure Machine Learning workspace |
+| `azurerm_key_vault` | An Azure Key Vault instance associated to the Azure Machine Learning workspace |
+| `azurerm_storage_account` | An Azure Storage instance associated to the Azure Machine Learning workspace |
+| `azurerm_container_registry` | An Azure Container Registry instance associated to the Azure Machine Learning workspace |
+| `azurerm_machine_learning_workspace` | An Azure Machine Learning workspace instance |
+| `azurerm_virtual_network` | An Azure Machine Learning workspace instance |
+| `azurerm_subnet` | An Azure Machine Learning workspace instance |
+| `azurerm_private_endpoint` | Private Endpoints for the Azure Machine Learning workspace and associated resources |
+| `azurerm_machine_learning_compute_instance` | An Azure Machine Learning compute instance a single-node managed compute. |
+| `azurerm_machine_learning_compute_cluster` | An Azure Machine Learning compute cluster as multi-node shared and managed compute. |
+| `azurerm_network_security_group` | Network security group with required inbound and outbound rules for Azure Machine Learning. |
+
+
+## Variables
+
+| Name | Description |
+|-|-|
+| name | Name of the deployment |
+| environment | The deployment environment name (used for pre- and postfixing resource names) |
+| location | The Azure region used for deployments |
+| image_build_compute_name | Name of the compute cluster to be created and configured for building docker images (Azure ML Environments)  |
+| training_subnet_resource_id | Resource ID of the existing training subnet |
```

Comment:
All of the *_resource_id variables should be handled differently so users will not need to pass in a bunch of big convoluted strings. Can the variables take names for these instead and use [data sources](https://www.terraform.io/docs/language/data-sources/index.html) to get the required id in the resource itself? LMK if you need examples, I am happy to help.

Reply:
@grayzu Are there any lean ways of doing this code-wise, to allow for flexibility to a situation where e.g. private DNS zones may reside in a different subscription opposed to the rest of the resources? Its seeking a bit of a balance between user-friendliness and comprehensiveness of the example.

Reply:
We pushed some changes to use the data construct for existing networks and subnets. Where we didn't use the data construct is private link DNS zones. Since they may come from the same subscription, another subscription or multiple subscriptions and/or resource groups, we think it would make the example a bit complicated.